roadservice/core/models.py

```diff
@@ -233,13 +293,93 @@ def __init__(self, *args, **kwargs):
         return self
 
+    def add_service_team(self, speciality, team_members_users):
+        for team_member in team_members_users:
+            if team_member.has_role():
+                return None
+        new_team = ServiceTeam.objects.create(county=self.county, speciality=speciality)
+        for team_member in team_members_users:
+            Serviceman.objects.create(team=new_team, user=team_member)
+        return new_team
+
+    def edit_service_team(self, team, speciality, team_members_users):
+        ex_members_users = []
+        for member in team.members.all():
+            ex_members_users.append(member.user)
+        for user in team_members_users:
+            if user.has_role() and user not in ex_members_users:
+                return False
+        for user in team_members_users:
+            if user.has_role() and user in ex_members_users:
+                continue
+            Serviceman.objects.create(user=user, team=team)
+        for member in team.members.all():
+            if member.user not in team_members_users:
+                member.delete()
+        team.speciality = speciality
+        return True
+
+    def delete_service_team(self, team):
+        for member in team.members.all():
+            member.delete()
+        team.delete()
+
+    def add_speciality(self, name):
+        if not Speciality.objects.filter(name=name).exists():
+            Speciality.objects.create(name=name)
+        return Speciality.objects.get(name=name)
+
+    def rename_speciality(self, speciality, new_name):
+        speciality.name = new_name
+        speciality.save()
+
+    def delete_speciality(self, speciality):
+        speciality.delete()
+
+    def pre_assign_expert(self, user):
+        # Check whether the operation is valid and dismiss the previous expert
+        if self.county.has_expert() and self.county.expert.user == user:
+            raise Exception('The user is already the expert of the county')
+        elif user.has_role():
+            raise Exception('The user has a role')
+        elif self.county.has_expert():
+            self.county.expert.delete()
+
+    def assign_expert(self, user):
+        self.pre_assign_expert(user)
+        return CountyExpert.objects.create(user=user, county=self.county)
+
+    def increase_machinery(self, machine_type):
+        if not Machinery.objects.filter(type=machine_type).exists():
+            Machinery.objects.create(type=machine_type, total_count=0, available_count=0, county=self.county)
+        machine = Machinery.objects.get(type=machine_type)
+        machine.total_count += 1
+        machine.available_count += 1
+        machine.save()
+        return machine
+
+    def decrease_machinery(self, machine_type):
+        if not Machinery.objects.filter(type=machine_type).exists():
+            return False
+        machine = Machinery.objects.get(type=machine_type)
+        if machine.available_count >= 1:
+            machine.available_count -= 1
```

Comment:
I think it's better to make a method for `Machinery` consisting these two lines. For both increasing, and decreasing.